File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/CompletePhoneNumberAuth.java
package me.grishka.houseclub.api.methods;

import me.grishka.houseclub.api.ClubhouseAPIRequest;
import me.grishka.houseclub.api.model.User;

public class CompletePhoneNumberAuth extends ClubhouseAPIRequest<CompletePhoneNumberAuth.Response> {

    public CompletePhoneNumberAuth(String phoneNumber, String code) {
        super("POST", "complete_phone_number_auth", Response.class);
        requestBody = new Body(phoneNumber, code);
    }

    public static class Response {
        public String authToken, accessToken, refreshToken;
        public boolean isWaitlisted;
        public User userProfile;
    }

    private static class Body {
        public String phoneNumber, verificationCode;

        public Body(String phoneNumber, String verificationCode) {
            this.phoneNumber = phoneNumber;
            this.verificationCode = verificationCode;
        }
    }
}

File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/CompletePhoneNumberAuth.kt
package me.grishka.houseclub.api.methods

import me.grishka.houseclub.api.ClubhouseAPIRequest
import me.grishka.houseclub.api.model.User

class CompletePhoneNumberAuth(phoneNumber: String, code: String) :
    ClubhouseAPIRequest<CompletePhoneNumberAuth.Response?>("POST", "complete_phone_number_auth", Response::class.java) {
    class Response {
        var authToken: String? = null
        var accessToken: String? = null
        var refreshToken: String? = null
        var isWaitlisted = false
        var userProfile: User? = null
    }

    private class Body(var phoneNumber: String, var verificationCode: String)

    init {
        requestBody = Body(phoneNumber, code)
    }
}